src/Microsoft.AspNet.OData/EnableQueryAttribute.cs

```diff
@@ -71,6 +73,37 @@ public override void OnActionExecuted(HttpActionExecutedContext actionExecutedCo
                 throw Error.Argument("actionExecutedContext", SRResources.ActionExecutedContextMustHaveActionContext);
             }
 
+            bool hasExpandQuery = false;
+            string queryString = request.RequestUri.Query;
+            var absoluteUri = request.RequestUri.AbsoluteUri;
+            bool hasEmptyQueryString = string.IsNullOrEmpty(queryString);
+            string baseUriWithPath = hasEmptyQueryString ? absoluteUri : absoluteUri.Substring(0, absoluteUri.IndexOf('?', 0));
```

Comment:
I think we could have used a `ReadOnlySpan<char>` here to avoid the `Substring` allocation. But I'm not sure if it's supported on all our target frameworks.

Reply:
Fixed